{{ cookiecutter.__project_name_kebab }}/.pre-commit-config.yaml

```diff
@@ -26,23 +26,49 @@ repos:
         args: ['--target-version', 'py37']
   - repo: https://github.com/pycqa/isort
     # isort config is in setup.cfg
-    rev: 5.10.1
+    rev: 5.12.0
     hooks:
       - id: isort
   - repo: https://github.com/pycqa/flake8
     # flake8 config is in setup.cfg
-    rev: 5.0.4
+    rev: 6.0.0
     hooks:
       - id: flake8
         additional_dependencies:
           - flake8-bugbear
           - flake8-comprehensions
   - repo: https://github.com/pre-commit/mirrors-prettier
-    # note: keep in sync with the version from package.json
-    rev: 'v1.18.2'
+    # prettier config is in prettier.config.js
+    rev: 'v2.7.1'
     hooks:
       - id: prettier
         types_or: [css, scss, javascript, ts, tsx, json, yaml]
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    # eslint config is in .eslintrc.js
+    rev: v8.32.0
+    hooks:
+      - id: eslint
+        additional_dependencies:
+          - 'eslint@8.32.0'
+          - 'eslint-config-airbnb@19.0.4'
+          - 'eslint-plugin-import@2.27.5'
+          - 'eslint-plugin-jsx-a11y@6.7.1'
+          - 'eslint-plugin-react@7.32.1'
+          - 'eslint-plugin-react-hooks@4.6.0'
+          - '@typescript-eslint/eslint-plugin@5.49.0'
+          - '@typescript-eslint/parser@5.49.0'
+          - '@wagtail/eslint-config-wagtail@0.4.0'
```

Comment:
I don't think this is a blocker, but another pre-commit question as I continue my odyssey of trying to get on-board with it:
It seems like pre-commit requires the user to resolve their own dependency tree here. Whereas with NPM we can just specify the top-level plugins we care about (`@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser`, `@wagtail/eslint-config-wagtail`) and leave NPM to resolve the transitive dependencies, that doesn't work with pre-commit. It seems you need to explicitly specify each plugin/config referenced in eslintrc in `.pre-commit-config.yaml` (this is what we do in WTKit too for reference). I was able to work it out by npm install-ing everything into a clean project somewhere else, letting NPM resolve the constraints and then using `npm list` to extract the packages/version numbers from the resolved tree to manually plug back in to the yaml file, but this doesn't feel like an ideal workflow to impose on package authors as they maintain/update this over time. It is not something I would ideally choose to live with on an ongoing basis. Am I _doing it wrong_? Is there a better solution for this that exists in pre-commit land?. This seems to be a characteristic which is mostly ESLint-specific as a typical ESLint config relies on many third-party plugins and configs, which is not the case for most other tools we use.